narwhals/_pandas_like/series_dt.py

```diff
@@ -190,3 +192,39 @@ def timestamp(self, time_unit: TimeUnit) -> PandasLikeSeries:
             raise TypeError(msg)
         result[mask_na] = None
         return self.with_native(result)
+
+    def truncate(self, every: str) -> PandasLikeSeries:
+        multiple, unit = parse_interval_string(every)
+        native = self.native
+        dtype_backend = get_dtype_backend(native.dtype, self.compliant._implementation)
+        if unit in {"mo", "q", "y"}:
+            if self.implementation.is_cudf():
+                msg = f"Truncating to {unit} is not supported yet for cuDF."
+                raise NotImplementedError(msg)
+            if dtype_backend == "pyarrow":
+                import pyarrow.compute as pc  # ignore-banned-import
+
+                from narwhals._arrow.utils import UNITS_DICT
+
+                ca = native.array._pa_array
+                result_arr = pc.floor_temporal(ca, multiple, UNITS_DICT[unit])
+            else:
+                if unit == "q":
+                    multiple *= 3
+                    np_unit = "M"
+                elif unit == "mo":
+                    np_unit = "M"
+                else:
+                    np_unit = "Y"
+                arr = native.values
+                arr_dtype = arr.dtype
+                result_arr = arr.astype(f"datetime64[{multiple}{np_unit}]").astype(
+                    arr_dtype
+                )
+            result_native = native.__class__(
+                result_arr, dtype=native.dtype, index=native.index, name=native.name
+            )
+            return self.with_native(result_native)
+        if unit == "m":  # replace "m" unit with "min"
+            every += "in"
```

Comment:
When comments like this were needed, it's usually a good sign to write something different 😉 

```suggestion
        if unit == "m":
            every = every.replace("m", "min", count=1)
```